src/datasets/features.py

```diff
@@ -129,7 +172,7 @@ def cast_to_python_objects(obj: Any) -> Any:
 class Value:
     """Encapsulate an Arrow datatype for easy serialization."""
 
-    dtype: str
+    dtype: str  # The string representation of a pyarrow datatype: str(pyarrow.DataType)
```

Comment:
From the documentation here: https://huggingface.co/docs/datasets/features.html

I think it would be helpful to clarify whether this `dtype` should be the name of a pyarrow function, or if it should be the string representation of a pyarrow DataType:

i.e. `float64` or `double`?

My intuition is that if you want to accept `dtype`s such as `timestamp[ns]`, then this dtype value should correspond to `str(pa.DataType)` instead of `pyarrow.__dict__[<factory function name>]` - otherwise, you have one style of dtype for certain primitives, and another altogether for timestamps, decimal, etc.:

```
>>> str(pyarrow.decimal128(1))
'decimal(1, 0)'
```

I realize that double->float64 and float->float32 are already on the public interface so I wouldn't propose changing those since that would be a backward-incompatible change, but it would be useful for documenting going forward, I think.

Happy to accept a decision either way and to remove this comment, but wanted to highlight it so that the maintainers can provide explicit guidance.  Either way you choose, a comment and some additional documentation on the https://huggingface.co/docs/datasets/features.html page would probably be helpful, and I'd be happy to help with that.

Reply:
If you want to maintain your own dtypes separate from pyarrow, then I think that makes a lot of sense.

In that world, perhaps there should actually be a separate function `arrow_to_string` that would act as an inverse for `string_to_arrow` that you could specifically control vs. relying on whatever `str(pa_type)` happens to do?

I'm happy to clean up my PR with this in mind, and I can open a follow-up PR that will incorporate this suggestion and add some documentation if you'd like!

Reply:
It makes sense to have `arrow_to_string` indeed, thanks for the suggestion and for the help :)
And sure feel free to add some documentation, this is greatly appreciated

We try to have our own dtypes so that we can provide alternative backends instead of pyarrow, which can be a bit overkill for data that easily fit in memory IMO. This is something we're exploring